src/ui/common/src/components/integrations/integrationObjectPreview.tsx

```diff
@@ -3,40 +3,62 @@ import Box from '@mui/material/Box';
 import CircularProgress from '@mui/material/CircularProgress';
 import React from 'react';
 
-import DataTable from '../../components/tables/dataTable';
 import { ObjectState } from '../../reducers/integration';
 import { isFailed, isLoading, isSucceeded } from '../../utils/shared';
+import VirtualizedTable from '../tables/virtualizedTable';
 
 type Props = {
   objectName: string;
   object: ObjectState;
 };
 
 const IntegrationObjectPreview: React.FC<Props> = ({ objectName, object }) => {
-  return (
-    <Box sx={{ mt: 3 }}>
-      {isLoading(object.status) && (
-        <Box sx={{ display: 'flex', flexDirection: 'row', mt: 3 }}>
-          <CircularProgress size={30} />
-          <Typography sx={{ ml: 2 }}>
-            Loading object <b>{objectName}</b>...
-          </Typography>
-        </Box>
-      )}
-      {isFailed(object.status) && (
-        <Alert style={{ marginTop: '10px' }} severity="error">
-          Object <b>{objectName}</b> failed to load. Try refreshing the page.{' '}
-          <br />
-          Error: {object.status.err}
-        </Alert>
-      )}
-      {isSucceeded(object.status) && !!object.data && (
-        <Box sx={{ height: '50vh', width: '100%' }}>
-          <DataTable data={object.data} />
-        </Box>
-      )}
-    </Box>
-  );
+  let content: React.ReactElement;
+  if (isLoading(object.status)) {
+    content = (
+      <Box sx={{ display: 'flex', flexDirection: 'row', mt: 3 }}>
+        <CircularProgress size={30} />
+        <Typography sx={{ ml: 2 }}>
+          Loading object <b>{objectName}</b>...
+        </Typography>
+      </Box>
+    );
+  }
+  if (isFailed(object.status)) {
+    content = (
+      <Alert style={{ marginTop: '10px' }} severity="error">
+        Object <b>{objectName}</b> failed to load. Try refreshing the page.{' '}
+        <br />
```

Comment:
Rather than adding a `<br />` here, lets instead put the above title in an `<AlertTitle>` component and then have the error message be the body of the error.

Reply:
As a general rule, we really shouldn't be using <br /> and should be using css margins/paddings instead.